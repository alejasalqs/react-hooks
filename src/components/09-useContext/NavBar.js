import React from "react";
import { NavLink } from "react-router-dom";
// La diferencia entre Link y NavLink es que con NavLink nos permite saber en cual ruta estamos para
// activar un clase de css, en el menÃº se ve el ejemplo usando el atributo activeClassName
export const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-light bg-light">
      <div className="container-fluid">
        <NavLink className="navbar-brand" to="usecontext/">
          useContext Hook
        </NavLink>
        <button
          className="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarNavAltMarkup"
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon"></span>
        </button>
        <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
          <div className="navbar-nav">
            <NavLink
              exact
              activeClassName="active"
              className="nav-link active"
              aria-current="page"
              to="usecontext/"
            >
              Home
            </NavLink>
            <NavLink
              exact
              activeClassName="active"
              className="nav-link active"
              aria-current="page"
              to="usecontext/about"
            >
              About
            </NavLink>
            <NavLink
              exact
              activeClassName="active"
              className="nav-link active"
              aria-current="page"
              to="usecontext/login"
            >
              Login
            </NavLink>
          </div>
        </div>
      </div>
    </nav>
  );
};
