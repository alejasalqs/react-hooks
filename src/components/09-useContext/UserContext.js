import { createContext } from "react";
// hire order component
export const UserContext = createContext(null);
