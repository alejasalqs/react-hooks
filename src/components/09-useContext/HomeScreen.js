import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {
  // Obtenemos la instancia del tipo de context que mandemos por parametro
  // como es un objeto se puede desestructurar
  const { user, setUser } = useContext(UserContext);

  return (
    <>
      <h1>Home</h1>
      <hr />

      <pre>{JSON.stringify(user)}</pre>
    </>
  );
};
