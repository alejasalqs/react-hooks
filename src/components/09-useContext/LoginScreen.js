import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);

  const user = {
    id: 123,
    name: "Alejandro Salguero",
    email: "alejandro@mail.com",
  };

  return (
    <>
      <h1>Login</h1>
      <hr />

      <button className="btn btn-primary" onClick={() => setUser(user)}>
        Login
      </button>
    </>
  );
};
