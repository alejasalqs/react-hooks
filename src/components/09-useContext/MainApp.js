import React, { useState } from "react";
import { AppRouter } from "./Router/AppRouter";
import { UserContext } from "./UserContext";

export const MainApp = () => {
  // el .provider nos permite compartir info con los componentes hijos
  // El provider notifica a los hijos si hay cambios
  const [user, setUser] = useState({});

  // para mandar varias parametros se manda un objeto
  return (
    <UserContext.Provider value={{ user, setUser }}>
      <AppRouter />
    </UserContext.Provider>
  );
};
