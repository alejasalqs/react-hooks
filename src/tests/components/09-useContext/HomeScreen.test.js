import React from "react";
import Adapter from "enzyme-adapter-react-16";
const { shallow, configure } = require("enzyme");
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

configure({ adapter: new Adapter() });
describe("Pruebas en el componente <HomeScreen />", () => {
  const user = {
    name: "Alejandro",
    email: "alejandro@test.com",
  };
  // En este caso es mejor usar mount
  // Pero solo es compatible con react 16 y este es 17
  const wrapper = shallow(
    // Como el componente necesita un useContext hacemos eso de esta manera
    <UserContext.Provider value={user}>
      <HomeScreen />
    </UserContext.Provider>
  );
  test("Debe de mostrarse correctmente ", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
