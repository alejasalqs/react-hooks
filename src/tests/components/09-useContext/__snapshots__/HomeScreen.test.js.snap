// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <HomeScreen /> Debe de mostrarse correctmente  1`] = `ShallowWrapper {}`;
